apps/condo/domains/billing/schema/helpers/categoryResolver.js

```diff
@@ -0,0 +1,74 @@
+const { isEmpty, get } = require('lodash')
+
+const { generateGqlQueries } = require('@open-condo/codegen/generate.gql')
+const { generateServerUtils } = require('@open-condo/codegen/generate.server.utils')
+
+const { DEFAULT_BILLING_CATEGORY_ID } = require('@condo/domains/billing/constants/constants')
+const { loadListByChunks } = require('@condo/domains/common/utils/serverSchema')
+
+const BILLING_CATEGORY_FIELDS = '{ id name serviceNames }'
+const BillingCategoryGQL = generateGqlQueries('BillingCategory', BILLING_CATEGORY_FIELDS)
+const BillingCategoryApi = generateServerUtils(BillingCategoryGQL)
+
+class CategoryResolver {
```

Comment:
Why do we need a class for this? 

The only thing that is stored in `this` is categories, which can be fetched from DB / JSON anyways, why not make a module with static functions? It would be easier to read and use

I would get it if CategoryResolver were a singletone, or if Resolver pattern is common in project, but I'm not sur on this.

It also would be really cool if you added docstrings

Reply:
Because, there will be a single class for every instance the unified mutation will work: Category, Account, Recipient, Property, Receipt. They all will have base class for logging, error handling and caching mechanics. Work with receipt will be like using the chain of classes. And yes, they will be singleton . To separate logic to different classes is better then to have one super-function 5 pages in length. It will be also easy to write tests for them as they have predictable input and needs to produce predictable output.

Reply:
I prefer to use that way to handle category resolve logic, since we have state in code and we have methods that manage/read state